### ISSUE TITLE
Invalid filesystem when using File Upload field

### ISSUE THREAD

**Issue by angelokeirsebilck** [verbb/formie#2188]
### Describe the bug

I'm using a File Upload field with no Upload Location selected and no path given.

Submitting the form throws an error:

![image](https://github.com/user-attachments/assets/b6f32b19-7e31-4e4e-9a5f-5ee5d459e60f)

I've tried selecting an Upload Location but then it throws another error when submitting, the Upload Location is not used in the path location of the file.

![image](https://github.com/user-attachments/assets/259956ed-58c0-4c6a-bc1e-c2edd4167b65)

![image](https://github.com/user-attachments/assets/09bf3b30-d3b9-49ec-8c25-9b05b01fca6e)

In this example i've set the Upload Location to Files but it is not added to the path of the pdf.

### Steps to reproduce

1. Add a File Upload field
2. Do not select a Upload Location
3. Submit the form
4. Check if it throws an error
5. Add a Upload Location
6. Submit the form
7. Check if it throws an error

### Form settings

- Multi-page form: No
- Submission Method: Server Side

### Craft CMS version

Craft Pro 5.5.5

### Plugin version

3.0.14

### Multi-site?

Yes

**Comment by engram-design** [comment]
> I'm using a File Upload field with no Upload Location selected and no path given.

You'll need to set at least the "Upload Location" for file uploads to work.

Can you let me know the details for that "Files" volume and file system used? It [should](https://github.com/verbb/formie/blob/fe8b825880b126af8cae682856957e11331c448a/src/fields/FileUpload.php#L914-L977) be resolving to the path set for that volume.

**Comment by angelokeirsebilck** [comment]
> > I'm using a File Upload field with no Upload Location selected and no path given.
> 
> You'll need to set at least the "Upload Location" for file uploads to work.
> 
> Can you let me know the details for that "Files" volume and file system used? It [should](https://github.com/verbb/formie/blob/fe8b825880b126af8cae682856957e11331c448a/src/fields/FileUpload.php#L914-L977) be resolving to the path set for that volume.

Is this new in Formie v3, can't remember I had to select this before.

The Filesystem:

![image](https://github.com/user-attachments/assets/5dd67392-202c-4d00-9fb5-00726fe79db9)

The volume:

![image](https://github.com/user-attachments/assets/48d203fb-8555-423a-914b-5bcc27ee8d05)

Aliases:

```php
    ->aliases([
        '@webroot' => dirname(__DIR__) . '/web',
        '@web' => craft\helpers\App::env('PRIMARY_SITE_URL'),
    ])
``` 

With PRIMARY_SITE_URL just the default base url of the website.

**Comment by engram-design** [comment]
Yep, you've always needed to select where to upload the file for as long as the field has existed I believe.

Those settings like fine to me, the same as what I'm using in fact.

I have seen an issue where there is some sort of reactivity issue in the select field's UI, where even though set, the upload location isn't set properly. This can be sorted by picking another volume, then switching back. Not sure if that'll help (still trying to figure out what's causing that), but worth a shot.

**Comment by angelokeirsebilck** [comment]
> Yep, you've always needed to select where to upload the file for as long as the field has existed I believe.
> 
> Those settings like fine to me, the same as what I'm using in fact.
> 
> I have seen an issue where there is some sort of reactivity issue in the select field's UI, where even though set, the upload location isn't set properly. This can be sorted by picking another volume, then switching back. Not sure if that'll help (still trying to figure out what's causing that), but worth a shot.

Changing the volume did not help but I did found something else. The actual file is saved to the right place and is in fact available in the submission. However the path used to attach this file in the e-mail is wrong, that is where the error is coming from.

The volume path, in my case '/files' is not in the path.

**Comment by engram-design** [comment]
Aha, yes you're right about that - it's the email notification attachment throwing that issue. Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.14"`.

**Comment by engram-design** [comment]
Fixed in [3.0.15](https://github.com/verbb/formie/releases/tag/3.0.15)
